src/debugger/ios/iOSPlatform.ts

```diff
@@ -76,13 +77,36 @@ export class IOSPlatform implements IAppPlatform {
             return Q.resolve<void>(void 0);
         }
 
-        return new IOSDebugModeManager(this.projectPath).setSimulatorJSDebuggingModeSetting(/*enable=*/ true)
-            .then(() => {
-                return this.plistBuddy.getBundleId(launchArgs.projectRoot);
-            }).then((bundleId: string) => {
-                // Relaunch the app so the new setting can take effect
-                return new CommandExecutor().execute(`xcrun simctl launch booted ${bundleId}`);
-            });
+        const iosDebugModeManager = new IOSDebugModeManager(this.projectPath);
+
+        // Wait until the configuration file exists, and check to see if debugging is enabled
+        return Q.all([
+            iosDebugModeManager.getSimulatorJSDebuggingModeSetting(),
+            this.plistBuddy.getBundleId(launchArgs.projectRoot)
+        ]).spread((debugModeSetting: string, bundleId: string) => {
+            if (debugModeSetting !== IOSDebugModeManager.WEBSOCKET_EXECUTOR_NAME) {
+                // Debugging must still be enabled
+                const commandExecutor = new CommandExecutor();
+                const childProcess = new ChildProcess();
+                const launchAppString = `xcrun simctl launch booted ${bundleId}`;
+                // simctl launch returns the process ID of the app in the simulator
+                return childProcess.exec(launchAppString).outcome.then((buffer: Buffer) => {
+                    // Kill the simulated app
+                    const pidMatch = buffer.toString().match(/: ([0-9]+)/);
+                    const pid = pidMatch[1];
+                    return commandExecutor.execute(`kill ${pid} # Restarting app`);

```

Comment:
Ah, I was not aware of that command. It seems to work much better and means we do not need the arbitrary delay any more I think. I'll test it some more to be sure. Thanks!